cmd/rook/mon.go

```diff
@@ -88,3 +95,35 @@ func startMon(cmd *cobra.Command, args []string) error {
 
 	return nil
 }
+
+// Compare the expected mon keyring secret with the cached keyring from a previous run of the monitor.
+// If these don't match we will not want to launch the monitor.
+func compareMonSecret(secret, configDir string) error {
+	cachedKeyringFile := path.Join(configDir, "data", "keyring")
+	if _, err := os.Stat(cachedKeyringFile); os.IsNotExist(err) {
+		// the mon is starting for the first time
+		logger.Infof("mon keyring is not yet cached at %s", cachedKeyringFile)
+		return nil
+	}
+
+	contents, err := ini.Load(cachedKeyringFile)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to read config file %s. %+v. Skipping check for cached keyring.\n", cachedKeyringFile, err)
+		return nil
```

Comment:
should you return the error here? and also down below where you are getting an error.

Reply:
i wanted to be conservative and only fail in the exact condition that is being checked. If there is still an error condition, it will fail later.